app/code/Magento/Backend/Model/Url.php

```diff
@@ -166,7 +167,10 @@ protected function _isSecure()
         if ($this->hasData('secure_is_forced')) {
             return $this->getData('secure');
         }
-        return $this->_scopeConfig->isSetFlag('web/secure/use_in_adminhtml');
+        if ($this->_scopeConfig->isSetFlag('web/secure/use_in_adminhtml')) {
```

Comment:
this change would currently make the setting `web/secure/use_in_adminhtml` = 0 ineffective

Reply:
> when he sets non secure connection at the first magento setup, and then he wants to secure it, how should he proceed?

I already answered on this here: https://github.com/magento/magento2/issues/11356#issuecomment-335609411

> what if something goes not well?

You can apply recipe from my comment as many times as you want.